src/EditorFeatures/Core/Implementation/TodoComment/TodoCommentIncrementalAnalyzer.cs

```diff
@@ -56,38 +57,49 @@ public async Task AnalyzeSyntaxAsync(Document document, InvocationReasons reason
                 return;
             }
 
+            // Compute and persist the TODO comments for this document.
+            var (existingData, newData) = await ComputeAndPersistTodoCommentsAsync(document, _state, _todoCommentTokens, cancellationToken).ConfigureAwait(false);
+
+            // Now update the task list if there are any changes from the previously computed TODO comments, if any.
+            // NOTE: We don't check for cancellation here to ensure the task list view and our latest
+            // computed state are identical.
+            if (existingData == null || existingData.Items.Length != newData.Items.Length)
```

Comment:
Can existingData and newData has different items but same length?

Reply:
Good point. However, this hasn't change as part of this PR, I am just refactoring the code in this PR. I will play around a bit with this and see if I can get a repro to drive this change.